### PR TITLE
feat: notify maintainers when recommended issue queue is empty

### LINKED ISSUE #1413: [Beginner]: Ping maintainers when no recommended issues are available after PR merge
### 🐥 Beginner Friendly

This issue is a great fit for contributors who are ready to explore the Hiero C++ codebase a little more and take on slightly more independent work.

Beginner Issues often involve reading existing code, understanding how different parts of the SDK fit together, and making small, thoughtful updates that follow established patterns.

The goal is to support skill growth while keeping the experience approachable, well-scoped, and enjoyable.

> [!IMPORTANT]
> ### 🐥 About Beginner Issues
>
> Beginner Issues are a great next step for contributors who feel comfortable with the basic project workflow and want to explore the codebase a little more.
>
> These issues often involve:
> - Reading existing code  
> - Understanding how different parts of the SDK fit together  
> - Making small, thoughtful updates that follow established patterns  
>
> You'll usually see Beginner Issues focused on things like:
> - Small, well-scoped improvements to existing tests  
> - Narrow updates to existing functionality (e.g. refining helpers or improving readability)  
> - Documentation or comment clarity  
> - Enhancements to existing examples  
>
> Other types of contributions — such as brand-new features, broader system changes, or deeper technical work — are just as valuable and may use different labels.

### 👾 Description of the Task

When a contributor merges a PR, the bot runs `handleRecommendIssues` in `.github/scripts/commands/recommend-issues.js` to suggest next issues to work on. Currently, when no suitable issues are found (the `issues.length === 0` case), the bot silently returns without posting any comment. The contributor receives no feedback at all — a missed opportunity to keep them engaged and to signal to maintainers that the issue queue needs replenishment.

When no issues are available, the bot should post a comment on the PR tagging the maintainer team (`@hiero-ledger/hiero-sdk-cpp-maintainers`) to let them know more issues are needed at the appropriate skill level.

Relevant files:
```
.github/scripts/commands/recommend-issues.js
.github/scripts/tests/test-recommend-issues-bot.js
```

### 💡 Proposed Approach

Add a new comment builder function, `buildMaintainerNudgeComment(username, nextLevel)`, to `.github/scripts/commands/recommend-issues.js`. This function should return a message that:
- Acknowledges the contributor by `@username`
- Tags `MAINTAINER_TEAM`
- States that no issues are currently available at the target skill level
- Asks maintainers to create or finalize more issues at that level

In `handleRecommendIssues`, update the empty case (currently `if (issues.length === 0) { ... return; }`) to:
1. Compute `getNextLevel(skillLevel)` to determine the target level the contributor is ready for (this returns `skillLevel` itself when already at the highest level)
2. Call `postComment` with the result of `buildMaintainerNudgeComment`

`MAINTAINER_TEAM` is already imported at the top of `recommend-issues.js` — no new imports are needed.

### 👩‍💻 Implementation Steps

- [ ] Read `.github/scripts/commands/recommend-issues.js` to understand the existing `handleRecommendIssues` flow and the existing comment builder pattern (`buildRecommendationErrorComment`)
- [ ] Add a `buildMaintainerNudgeComment(username, nextLevel)` function to `recommend-issues.js` that:
  - Greets the contributor by `@username`
  - Tags `MAINTAINER_TEAM`
  - States that no open issues are currently available at the `nextLevel` skill level
  - Asks maintainers to create or finalize more issues at that level so the contributor has a clear next step
- [ ] In `handleRecommendIssues`, replace the silent `return` in the `issues.length === 0` branch with:
  - A call to `getNextLevel(skillLevel)` to determine the target level
  - A `postComment` call using `buildMaintainerNudgeComment(username, nextLevel)`
- [ ] Export `buildMaintainerNudgeComment` from `recommend-issues.js` (add it to the `module.exports` object at the bottom of the file)
- [ ] Add unit tests for `buildMaintainerNudgeComment` in `.github/scripts/tests/test-recommend-issues-bot.js` following the style of the existing comment builder tests in that file
- [ ] Add a test for the empty-results case in `handleRecommendIssues` verifying that the maintainer nudge comment is posted and no recommendation comment is posted
- [ ] Open a pull request with the change

### ✔️ Acceptance Criteria

- [ ] When a PR is merged and no suitable issues are found, the bot posts a comment on the PR tagging `@hiero-ledger/hiero-sdk-cpp-maintainers` asking them to create more issues at the appropriate skill level
- [ ] The comment includes the contributor's `@username` and the target skill level
- [ ] `buildMaintainerNudgeComment` is covered by unit tests
- [ ] The empty-results case in `handleRecommendIssues` is covered by a test verifying the nudge comment is posted and the success comment is not
- [ ] All existing tests continue to pass
- [ ] No unrelated behavior changes are introduced

---

### 📋 Step-by-Step Contribution Guide

To help keep contributions consistent and easy to review, we recommend following these steps:

- [ ] Comment `/assign` to request the issue
- [ ] Wait for assignment
- [ ] Fork the repository and create a branch
- [ ] Set up the project using the instructions in `README.md`
- [ ] Make the requested changes
- [ ] Sign each commit using `-s -S`
- [ ] Push your branch and open a pull request

Read [Workflow Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/workflow.md) for step-by-step workflow guidance.
Read [README.md](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/README.md) for setup instructions.

❗ Pull requests **cannot be merged** without `S` and `s` signed commits.
See the [Signing Guide](https://github.com/hiero-ledger/hiero-sdk-cpp/blob/main/docs/training/signing.md).

### 🤔 Additional Information

The existing `buildRecommendationErrorComment` function in `recommend-issues.js` already demonstrates how to tag `MAINTAINER_TEAM` in a bot comment — use it as a reference when writing `buildMaintainerNudgeComment`.

`MAINTAINER_TEAM` is already imported at the top of `recommend-issues.js` via the `helpers` module, so no additional imports are needed.

If you have questions while working on this issue, feel free to ask! [Hiero-SDK-C++ Discord](https://discord.com/channels/905194001349627914/1337424839761465364)

### PR DESCRIPTION
**Pull request by shady-2004**
**Description**:
Notify maintainers when the recommended issues queue is empty for a contributor's skill level. Previously, the bot would silently exit if no issues were found, leaving the contributor without guidance. 

* Add `buildMaintainerNudgeComment` to generate a formatted message tagging the maintainer team
* Update `handleRecommendIssues` to post a maintainer nudge comment instead of returning silently when recommendations are empty
* Add unit tests to verify the empty queue fallback logic and the new comment builder

**Related issue(s)**:

Fixes #1413

**Checklist**

- [x] Documented (Code comments)
- [x] Tested (unit)

### CODE REVIEW

File: .github/scripts/bot/bot-recommend-issues.js
Comment (rwalworth) [inline .github/scripts/bot/bot-recommend-issues.js:447]: **todo (blocking):** This JSDoc line is no longer accurate - the function no longer skips silently when no results are found.

Please update it to reflect the new behavior, for example:

```diff
- * Skips silently if context is incomplete or no results found.
+ * Skips silently if context is incomplete. Posts a maintainer nudge comment when no results are found.
```

File: .github/scripts/bot/bot-recommend-issues.js [hunk 1]
```diff
@@ -475,6 +494,9 @@ async function handleRecommendIssues(botContext) {
 
     if (issues.length === 0) {
         logger.log('recommendation.empty', { user: username });
+        const targetLevel = unlockedLevel || skillLevel;
+        const comment = buildMaintainerNudgeComment(username, targetLevel);
+        await postComment(botContext, comment);
```
Comment (rwalworth) [inline .github/scripts/bot/bot-recommend-issues.js:499]: **suggestion (non-blocking):** The `postComment` return value isn't checked here, which is inconsistent with the recommendation posting path below (which checks `postResult.success` and logs a failure). You could mirror that pattern:

```js
const postResult = await postComment(botContext, comment);
if (!postResult.success) {
    logger.error('recommendation.nudge.postCommentFailed', { error: postResult.error });
}
```

Not blocking - `postComment` already logs the error internally, so the failure won't go unnoticed.

File: .github/scripts/tests/test-recommend-issues-bot.js [hunk 2]
```diff
@@ -504,6 +505,19 @@ const unitTests = [
...
+  {
+    name: 'buildMaintainerNudgeComment: returns properly formatted message with team tag and nextLevel',
+    test: async () => {
+      const result = buildMaintainerNudgeComment('alice', LABELS.SKILL_BEGINNER);
+      return result.includes('@alice') &&
+             result.includes(MAINTAINER_TEAM) &&
+             result.includes(`**${LABELS.SKILL_BEGINNER}**`);
+    },
+  },
+
```
Comment (rwalworth) [inline .github/scripts/tests/test-recommend-issues-bot.js:520]: **issue (blocking):** The existing tests in this file use exact string snapshots when asserting comment content - for example, the `getRecommendedIssues` API failure test (around line 366) constructs the full expected string inline and compares with `===`. The new `buildMaintainerNudgeComment` test uses `.includes()` checks instead, which is inconsistent with that pattern and won't catch regressions in the comment's wording or structure.

Please rewrite it as a snapshot test:

```js
{
  name: 'buildMaintainerNudgeComment: returns properly formatted message with team tag and nextLevel',
  test: async () => {
    const result = buildMaintainerNudgeComment('alice', LABELS.SKILL_BEGINNER);
    const expected = [
      '👋 Hi @alice!',
      '',
      `It looks like there are currently no open issues available at the **${LABELS.SKILL_BEGINNER}** level.`,
      '',
      `${MAINTAINER_TEAM} — could you please create or finalize some issues at the **${LABELS.SKILL_BEGINNER}** level so they have a clear next step?`,
      '',
      'Thanks for your contributions!',
    ].join('\n');
    return result === expected;
  },
},
```

**Review by rwalworth — CHANGES_REQUESTED** [review]
Thanks for the contribution @shady-2004! The implementation looks great overall - you followed the existing pattern nicely and the tests cover both the new function and the updated `handleRecommendIssues` behavior. Just one test fix and small doc update needed before I can merge (see below).
